x-pack/plugins/reporting/public/lib/reporting_api_client.ts

```diff
@@ -7,59 +7,26 @@
 import { stringify } from 'query-string';
 import rison from 'rison-node';
 import { HttpSetup } from 'src/core/public';
-import { JobId, SourceJob } from '../../common/types';
+import { DownloadReportFn, ManagementLinkFn } from '../';
+import { JobId, ReportApiJSON, ReportDocument, ReportSource } from '../../common/types';
 import {
-  API_BASE_URL,
   API_BASE_GENERATE,
+  API_BASE_URL,
   API_LIST_URL,
   REPORTING_MANAGEMENT_HOME,
 } from '../../constants';
 import { add } from './job_completion_notifications';
 
 export interface JobQueueEntry {
   _id: string;
-  _source: any;
+  _source: ReportSource;
 }
 
 export interface JobContent {
   content: string;
   content_type: boolean;
 }
 
-export interface JobInfo {
-  kibana_name: string;
```

Comment:
This is the same data that `report.toApiJSON` returns, so I have refactored to maintain that.